### ISSUE TITLE
Error in .local(wf, action, ...) :    nodeID are not identical across samples!

### ISSUE THREAD

**Issue by mehrnoushmalek** [RGLab/flowWorkspace#239]
Hello,

I'm trying to create a gatingSet and then export to flowJo workspace. 
For some of the population I don't have enough cells for some samples, so I exclude these when adding the gate to gating set using add(). Then when I try to add another population I get this error, cause nodes are not identical anymore (i.e., some samples were excluded in add()) Is there any way around it, or is there a smarter way to do this? 
Basically I'd like to have a 20 cell threshold in order to do the gating. One of my sample has a low count on population A, but I don't want to exclude it when I'm gating population B, as it has more than 20 cells.
Maybe I'm doing something wrong. I'd be grateful if you can help me on this.

**Comment by gfinak** [comment]
A gatingset, by definition, has to have the same gating structure. You can get split your gating set into multiple groups each of which will have the same gates. 
There may be a "dummy gate" or something like that we use as a placeholder that gets around this, but have to check with @mikejiang.

**Comment by mikejiang** [comment]
Right, here is the example of `dummy gate` construction
https://github.com/RGLab/openCyto/blob/trunk/R/wrapper-functions.R#L45-L63

**Comment by mehrnoushmalek** [comment]
It makes sense. but when using FMOs, that could be an issue. Let say I have FMO of CD24 that I want to calculate its threshold based on memory populations (EM, CM, TEMRA and Naive). One sample might have only few cells for TEMRA, but more than enough for the other 3, and another sample only few for Naive, and enough for the rest. I don't want to exclude both from my gating structure.

**Comment by gfinak** [comment]
So, to put a fine point on it, @mehrnoushmalek, using @mikejiang example,  construct a rectangle gate in the dimensions you need with coordinates spanning -Inf to Inf in all dimensions and name it according to the missing population, and add it to the samples that are missing this population.

**Comment by mehrnoushmalek** [comment]
@mikejiang : This is actually similar what I'm doing right now, but I was secretly hoping this can be done when adding gates to gatingSet

**Comment by gfinak** [comment]
Are you doing data-driven gating?

**Comment by gfinak** [comment]
`minEvents <- openCyto.options[["gating"]][["minEvents"]]` 
setting minEvents might give you that behavior.  Not sure which gating methods it's attached to.

**Comment by mehrnoushmalek** [comment]
@gfinak Yes, but lots of FMOs are involved and I want to keep them in the gating strategy. They'd like to have counts for all populations on all samples that have more than 50 cells, and I wanted to avoid showing populations for these low counts population/samples.
Just one other thing:
On previous versions I had problem showing quad and rectangle gates in flowJo, is that fixed, or I don't know if you were aware of it.

**Comment by gfinak** [comment]
What I'm saying is that using the dummy gate, or setting minEvents will allow you to keep them in the same gatingSet.

**Comment by gfinak** [comment]
If you're having display issues with quad gates in flowJo, open an issue on RGLab/CytoML
